crates/store/re_tf/src/transform_forest.rs

```diff
@@ -277,31 +279,29 @@ impl TransformForest {
         };
 
         // Figure out the root frame for this entire stack.
-        let (mut root_frame, mut root_from_target) = if let Some(parent_from_child) =
-            top_of_stack.parent_from_child.as_ref()
+        let (mut root_frame, mut root_from_target) = if let Some(parent_frame) =
+            top_of_stack.parent_frame
         {
             // We have a connection further up the stack. That means we must have stopped because we already know that target!
-            if let Some(root_from_frame) = self.root_from_frame.get(&parent_from_child.parent) {
+            if let Some(root_from_frame) = self.root_from_frame.get(&parent_frame) {
                 // Yes, we can short-circuit to a known root!
                 debug_assert!(self.roots.contains_key(&root_from_frame.root));
                 (root_from_frame.root, root_from_frame.target_from_source)
             } else {
                 // We didn't know the target. Must mean that the target is a new root!
                 let previous_root = self.roots.insert(
-                    parent_from_child.parent,
+                    parent_frame,
                     // There's apparently no information about this root, so it can't be a pinhole!
                     TransformTreeRootInfo::TransformFrameRoot,
                 );
                 debug_assert!(previous_root.is_none(), "Root was added already"); // TODO(RR-2667): Build out into cycle detection
 
                 // That parent apparently won't show up in any transform stack (we didn't walk there because there was no information about it!)
                 // So if we don't add this root now to our `root_from_frame` map, we'd never fill out the required self-reference!
-                self.root_from_frame.insert(
-                    parent_from_child.parent,
-                    TransformInfo::new_root(parent_from_child.parent),
-                );
+                self.root_from_frame
+                    .insert(parent_frame, TransformInfo::new_root(parent_frame));
 
-                (parent_from_child.parent, glam::DAffine3::IDENTITY)
+                (parent_frame, glam::DAffine3::IDENTITY)
```

Comment:
here we always use the new `parent_frame` field of `ParentChildTransforms` so that we don't get it wrong!